### PR TITLE
Drone CI

### PR DESCRIPTION
**Pull request by phil-davis**
Issue #21 

I did the following things:
- committed the "standard" `.drone.star` that has all the "bells-and-whistles" used in CI for the oC10 apps. That gives flexibility to run whatever parts are wanted (code-style, phpunit, phan, phpstan...). And if stuff is fixed or enhanced in the "standard" `.drone.star` then the changes can be "mindlessly" applied the same here. (It gets annoying when some repos have different starlark code, and someone has to keep thinking about how to apply changes)
- enabled owncloud code-style v4 and committed the changes (it made format changes, and some `public constant` declarations)
- enabled phpunit - some of the unit tests need xdebug, so I adjusted `.drone.star` so that it installs php7.4-xdebug in the unit test pipeline. It uses its own locally-selected version of phpunit. The CI uses phpunit 9.6 with PHP 7.4. That decouples it from the unneeded core (many of the oC10 apps just run whatever phpunit version comes with core, we don't want to be hitched to that!)
- CI pipelines are run with PHP 7.4

Do we want the code-style changes? (if not,  I can remove those and not run code-style in CI)

Do we want to still support PHP 7.1 and up? Or drop the older PHP versions?

ToDo: sort out the unnecessary CI pipeline steps that install core... - this is not actually an oC10 app but I have used the "oC10 app" starlark code as an easy "template"

### CODE REVIEW

File: .phpunit.result.cache [hunk 1]
```diff
@@ -0,0 +1 @@
+{"version":1,"defects":{"ownCloud\\TarStreamer\\Tests\\Streamer::testSendHeadersOKWithRegularBrowser with data set #0":4,"ownCloud\\TarStreamer\\Tests\\Streamer::testSendHeadersOKWithRegularBrowser with data set #1":4,"ownCloud\\TarStreamer\\Tests\\Streamer::testSendHeadersOKWithRegularBrowser with data set #2":4,"ownCloud\\TarStreamer\\Tests\\Streamer::testSendHeadersOKWithRegularBrowser with data set #3":4},"times":{"ownCloud\\TarStreamer\\Tests\\Streamer::testSimpleFile with data set #0":0.017,"ownCloud\\TarStreamer\\Tests\\Streamer::testSimpleFile with data set #1":0.003,"ownCloud\\TarStreamer\\Tests\\Streamer::testAddingNoResource with data set #0":0.001,"ownCloud\\TarStreamer\\Tests\\Streamer::testAddingNoResource with data set #1":0.003,"ownCloud\\TarStreamer\\Tests\\Streamer::testDir":0.002,"ownCloud\\TarStreamer\\Tests\\Streamer::testSendHeadersOKWithRegularBrowser with data set #0":0,"ownCloud\\TarStreamer\\Tests\\Streamer::testSendHeadersOKWithRegularBrowser with data set #1":0,"ownCloud\\TarStreamer\\Tests\\Streamer::testSendHeadersOKWithRegularBrowser with data set #2":0,"ownCloud\\TarStreamer\\Tests\\Streamer::testSendHeadersOKWithRegularBrowser with data set #3":0}}
```
Comment (DeepDiver1975) [inline .phpunit.result.cache]: shall be excluded - right?

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

**Comment by DeepDiver1975** [comment]
You can fully drop all the steps regarding:
- install owncloud server
- install and test with different databases

This is just a tiny library - no owncloud app. see the travis.yaml as reference.

**Comment by phil-davis** [comment]
Yes, I pasted in the "standard" `.drone.star`. I will have a look at what to clean up when I get it running and passing.

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]
drop the databases -

**Comment by phil-davis** [comment]
> drop the databases -

done - yes, that made no sense at all

**Comment by DeepDiver1975** [comment]
mind adding php 8.0 and 8.1?

also having sqlite in the step name feels just wrong ...

**Comment by DeepDiver1975** [comment]
and there is still a step called `install core`

**Comment by DeepDiver1975** [comment]
also a step `setup-server-*`  is not neccessary

**Comment by phil-davis** [comment]
OK - I need to get rid of all the pieces related to core. I will have to cleanup all the logic of `.drone.star` so that it does not have all the "accidental" core stuff. We will have a custom cut-down `.drone.star`

**Comment by DeepDiver1975** [comment]
I can help out to make the code php8+ ready if you like..... In case there is anything to be done.

Thanks a lot so far....

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/TarStreamer/14/5/4
"This version of ownCloud is not compatible with PHP 8.0"

I really do need to get rid of anything that tries to use core!

**Comment by phil-davis** [comment]
@DeepDiver1975 I cleaned up a lot of the excess drone starlark.

I can easily enable phan and/or phpstan later if needed.

unit tests pass on PHP 7.4 8.0 8.1

anything else needed for this "first pass" at getting it going?

If this seems enough to get started, then I will squash some of the related commits together and then merge.

**Comment by DeepDiver1975** [comment]
Always interesting to see how an easy travis.yaml with 22 lines explodes into a 800+ lines drone ci file :see_no_evil: 

anyway :+1:

**Review by DeepDiver1975 — APPROVED** [review]

**Pull request merged**